Sources/Web3Core/Utility/String+Extension.swift

```diff
@@ -135,6 +135,26 @@ extension String {
     func trim() -> String {
         trimmingCharacters(in: .whitespacesAndNewlines)
     }
+
+   /// Splits a string into groups of `every` n characters, grouping from left-to-right by default. If `backwards` is true, right-to-left.
+   public func split(every: Int, backwards: Bool = false) -> [String] {
+       var result = [String]()
+
+       for i in stride(from: 0, to: self.count, by: every) {
+           switch backwards {
+           case true:
+               let endIndex = self.index(self.endIndex, offsetBy: -i)
+               let startIndex = self.index(endIndex, offsetBy: -every, limitedBy: self.startIndex) ?? self.startIndex
+               result.insert(String(self[startIndex..<endIndex]), at: 0)
+           case false:
+               let startIndex = self.index(self.startIndex, offsetBy: i)
+               let endIndex = self.index(startIndex, offsetBy: every, limitedBy: self.endIndex) ?? self.endIndex
+               result.append(String(self[startIndex..<endIndex]))
+           }
+       }
+
+       return result
+   }
```

Comment:
I believe this could be rewritten in the following way:

```swift
public func split(every: Int, backwards: Bool = false) -> [String] {
   var result = [String]()

   for i in stride(from: 0, to: self.count, by: every) {
        let endIndex = self.index(self.endIndex, offsetBy: backward ? -i : i)
        let startIndex = self.index(
            endIndex, 
            offsetBy: backward ? -every : every, 
            limitedBy: backward ? self.startIndex : self.endIndex
        ) ?? backward ? self.startIndex : self.endIndex
        result.insert(String(self[startIndex..<endIndex]), at: 0)
   }

   return result
}
```

But I pursue you to debate this one, coz on one hand it allows to avoid code duplication on the other it's a bit confusing.

Reply:
I was working through making the code I have in the format you suggest, I think my format will be more readable

Reply:
I tend to agree with that, but yet I'm worrying about code duplication.

Reply:
I think code duplication is acceptable in this case